kiwi/storage/disk.py

```diff
@@ -399,9 +401,14 @@ def map_partitions(self):
         required to map them if the storage provider is loop based
         """
         if self.storage_provider.is_loop():
-            Command.run(
-                ['partx', '--add', self.storage_provider.get_device()]
-            )
+            if self.partition_mapper == 'partx':
+                Command.run(
+                    ['partx', '--add', self.storage_provider.get_device()]
+                )
+            else:
+                Command.run(
+                    ['kpartx', '-s', '-a', self.storage_provider.get_device()]
+                )
```

Comment:
This condition should be flipped. Only when `kpartx` is used should we use `kpartx`. That makes it obvious that `kpartx` is the special case.

Reply:
done